Lib/unittest/case.py

```diff
@@ -300,24 +301,33 @@ def __exit__(self, exc_type, exc_value, tb):
 _LoggingWatcher = collections.namedtuple("_LoggingWatcher",
                                          ["records", "output"])
 
+# Import logging lazily for assertLogs().
+# assertLogs() is one of the least used assertions, and most tests do
+# not need to import logging.
 
-class _CapturingHandler(logging.Handler):
-    """
-    A logging handler capturing all (raw and formatted) logging output.
-    """
+def _lazy_logging_import():
```

Comment:
Hum. Would it be possible to move _AssertLogsContext class in a private unittest._log submodule and imports this one on demand?

Reply:
It's kind of unusal to define classes "only once" in a function. A separated submodule imported lazily sounds more common to me.

Reply:
Good idea! It looks clearer now.